docs/html/reference/inspect-report.md

```diff
@@ -0,0 +1,212 @@
+# `pip inspect` JSON output specification
+
+The `pip inspect` command produces a detailed JSON report of the Python
+environment, including installed distributions.
+
+## Specification
+
+The report is a JSON object with the following properties:
+
+- `version`: the string `0`, denoting that the inspect command is an experimental
+  feature. This value will change to `1`, when the feature is deemed stable after
+  gathering user feedback (likely in pip 22.3 or 23.0). Backward incompatible changes
```

Comment:
I wonder if we should commit to this… Versions numbers recorded in documentation likely this generally get stale without being updated.

Reply:
And, we're using "likely" to give us some leeway, in case we do miss this deadline in some way. :)